pyfmt/__init__.py

```diff
@@ -25,19 +28,139 @@
+        "              / _|         | |   ",
+        "  _ __  _   _| |_ _ __ ___ | |_  ",
+        " | '_ \| | | |  _| '_ ` _ \| __| ",
+        " | |_) | |_| | | | | | | | | |_  ",
+        " | .__/ \__, |_| |_| |_| |_|\__| ",
+        " | |     __/ |                   ",
+        " |_|    |___/                    ",
+    ]
+    print("\033[94m")
+    for part in pyfmt_splash:
+        print(f"{part.center(get_terminal_size().columns, ' ')}")
+    print("\033[0m")
+
+
+def find_all_files_and_dirs():
+    """
+    Map out all files and directories in the current
+    working directory
+    """
+    all_files = []
+    all_dirs = []
+    for root, dirs, files in os.walk("."):
+        for name in files:
+            # Saving filename
+            all_files.append(name)
+        for name in dirs:
+            # Saving directory name
+            all_dirs.append(os.path.abspath(os.path.join(root, name)))
+    # Remove duplicates
+    all_files = list(set(all_files))
+
+    return all_files, all_dirs
+
+
+def display_divider(title="", character="=", color_code="\033[94m"):
+    """
+    Divider between sections of program.
+    """
+    if title:
+        print(
+            "\n"
+            + "{}".format(color_code)
+            + "  {}  ".format(title.upper()).center(
+                get_terminal_size().columns, "{}".format(character)
+            )
+            + "\033[0m"
+        )
+    else:
+        print(
+            "\n"
+            + "{}".format(color_code)
+            + "".center(get_terminal_size().columns, "{}".format(character))
+            + "\033[0m"
+        )
+
+
+def pyfmt(
+    path, skip="", isort=False, black=False, check=False, line_length=100, show_title=False, extra_isort_args="", extra_black_args=""
+) -> int:
     """Run isort and black with the given params and print the results."""
+    if show_title:
+        pyfmt_title()  # Display title
+    timer_start = time.time()  # Measure how long everything takes
+
+    if skip:
+        # Map out current working directory
+        all_files, all_dirs = find_all_files_and_dirs()
+        # If specified files and directories exist, store the filenames
+        skips = skip.split(",")
+        filenames_to_skip = []
+        for item in skips:
+            if item in all_files:
+                if item.split(".")[-1] == "py":
+                    filenames_to_skip.extend([item])
+            elif os.path.abspath(item) in all_dirs:
+                # Saving all filenames in user specified directory
+                files_in_dir = []
+                for (dirpath, dirnames, filenames) in os.walk(item):
+                    for filename in filenames:
+                        if filename.split(".")[-1] in ("py", "pyi"):
+                            files_in_dir.append(filename)
+                filenames_to_skip.extend(files_in_dir)
+            else:
+                print(f'CRITICAL: One of the files or directories marked as skipped not found ("{item}").')
```

Comment:
if the user tried to skip it and it wasn't found, why is this a terminal condition? In either case, it won't run which is the desired behavior?

Reply:
it's a terminal condition because if the file was not found, more than likely the user made a spelling mistake, in which case the file they wanted to exclude got formatted.  I might be thinking about this wrong but I would figure I don't want specific files to be formatted by mistake...